src/main/scala/org/broadinstitute/dsde/snoop/ws/SnoopApiService.scala

```diff
@@ -17,21 +23,54 @@ object SnoopApiServiceActor {
   }
 }
 
-class SnoopApiServiceActor(override val executionServiceHandler: RequestContext => WorkflowExecutionService) extends Actor with SnoopApiService {
+class SnoopApiServiceActor(override val executionServiceHandler: RequestContext => WorkflowExecutionService) extends Actor with BaseSnoopApiService with WorkflowExecutionApiService {
+  implicit def executionContext = actorRefFactory.dispatcher
   def actorRefFactory = context
-  def receive = runRoute(snoopRoute)
-}
+  def possibleRoutes = baseRoute ~ workflowRoutes ~ swaggerService.routes 
 
+  def receive = runRoute(possibleRoutes)
 
-// this trait defines our service behavior independently from the service actor
-trait SnoopApiService extends HttpService {
-  implicit def executionContext = actorRefFactory.dispatcher
+  private val snoopConfig: Config = ConfigFactory.parseFile(new File("/etc/snoop.conf"))
+  private val swaggerConfig = snoopConfig.getConfig("swagger")
+
+  val swaggerService = new SwaggerHttpService {
+    override def apiTypes = Seq(typeOf[BaseSnoopApiService], typeOf[WorkflowExecutionApiService])
```

Comment:
Prefer using Vector as your concrete instantiation of Seq. Seq() ends up creating a List but Vector is generally more performant